### PR TITLE
[ADD] product_pricelist_assortment

### PR DESCRIPTION
**Pull request by benwillig**
@rousseldenis You might want to take a look

### CODE REVIEW

File: product_pricelist_assortment/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Product Pricelist Assortment",
+    "summary": """
+        Product assortment and pricelist""",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "ACSONE SA/NV, Odoo Community Association (OCA)",
+    "website": "https://acsone.eu",
```
Comment (rousseldenis) [inline product_pricelist_assortment/__manifest__.py]: Change to https://github.com/OCA/product-attribute

See https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: product_pricelist_assortment/models/product_pricelist_assortment_item.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    def _update_assortment_items(self):
+        """
+        Update the pricelist with current assortment:
+        - Prepare values for new assorment items;
+        - Delete previous items.
+        - Create new assortments items;
+
+        :return: bool
+        """
+        self.ensure_one()
```
Comment (rousseldenis) [inline product_pricelist_assortment/models/product_pricelist_assortment_item.py:82]: Make it as a real multi

File: product_pricelist_assortment/models/product_pricelist_assortment_item.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                self.display_name,
+            )
+            return False
+
+        PricelistItem = self.env["product.pricelist.item"].with_user(SUPERUSER_ID)
+        items_values = self._get_pricelist_item_values()
+        old_items = self._get_related_items()
+        old_items.unlink()
+        for item_value in items_values:
+            PricelistItem.create(item_value)
```
Comment (rousseldenis) [inline product_pricelist_assortment/models/product_pricelist_assortment_item.py]: Use create multi (list of values)

File: product_pricelist_assortment/models/product_pricelist.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        string="Assortment items",
+    )
+
+    def action_launch_assortment_update(self):
+        """
+        Action to execute update of assortment items
+        :return: dict
+        """
+        for item_assortment in self.mapped("item_assortment_ids"):
+            item_assortment._update_assortment_items()
```
Comment (rousseldenis) [inline product_pricelist_assortment/models/product_pricelist.py:23]: See comment here under about multi()

File: product_pricelist_assortment/models/product_pricelist_assortment_item.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class ProductPricelistAssortmentItem(models.Model):
+
+    _name = "product.pricelist.assortment.item"
+    _description = "Product Pricelist Assortment Item"
+    _inherit = "product.pricelist.item"
+
+    name = fields.Char(related="assortment_filter_id.name", readonly=True,)
```
Comment (rousseldenis) [inline product_pricelist_assortment/models/product_pricelist_assortment_item.py]: No need of readonly (default)

**Review by Cedric-Pigeon — APPROVED** [review]
LGTM. I would just fill CONTRIBUTOR.rst and DESCRIPTION.rst

**Review by acsonefho — APPROVED** [review]
LGTM!

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Some comments

**Comment by benwillig** [comment]
Updated the website key @rousseldenis, for the other requested changes, feel free to create a new PR or to push on this one

**Comment by Cedric-Pigeon** [comment]
@benwillig I think @rousseldenis proposal totally makes sense to speed up performances. It is not so complicated to do. Could you please reconsider?

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**